From 42b18c5a3d0f2ed93c9fe4ae3d2b3821d97d72f0 Mon Sep 17 00:00:00 2001
From: pzalewski-thegoodpenguin <pzalewski@thegoodpenguin.co.uk>
Date: Thu, 1 Dec 2022 11:04:12 +0000
Subject: [PATCH] Adding relative path to the LLVM toolchain

---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index f9d916f98..5e433677e 100644
--- a/Makefile
+++ b/Makefile
@@ -145,7 +145,7 @@ MORELLO_FLAGS		:= -Wno-cheri-inefficient -march=morello+c64
 ELF_PATCH		:= morello_elf
 MUSL_HOME		:= ../../musl-bin
 KHEADERS		:= ../morello-linux-headers/usr/include/
-CLANG_RESOURCE_DIR	:= $(shell clang -print-resource-dir)
+CLANG_RESOURCE_DIR	:= $(shell ${LLVM_PATH}/clang -print-resource-dir)
 
 export srctree objtree VPATH TOPDIR
 
@@ -296,16 +296,16 @@ MAKEFLAGS += -rR
 
 # Make variables (CC, etc...)
 
-AS		= llvm-as
-CC_		= clang
+AS		= ${LLVM_PATH}/llvm-as
+CC_		= ${LLVM_PATH}/clang
 CC		= $(CC_) --target=${MORELLO_TARGET} --sysroot ${MUSL_HOME} ${MORELLO_FLAGS}
 LD		= $(CC) -fuse-ld=lld -rtlib=compiler-rt
 CPP		= $(CC) -E
-AR		= llvm-ar
-NM		= llvm-nm
-STRIP		= llvm-strip
-OBJCOPY		= llvm-objcopy
-OBJDUMP		= llvm-objdump
+AR		= ${LLVM_PATH}/llvm-ar
+NM		= ${LLVM_PATH}/llvm-nm
+STRIP		= ${LLVM_PATH}/llvm-strip
+OBJCOPY		= ${LLVM_PATH}/llvm-objcopy
+OBJDUMP		= ${LLVM_PATH}/llvm-objdump
 PKG_CONFIG	?= $(CROSS_COMPILE)pkg-config
 AWK		= awk
 GENKSYMS	= scripts/genksyms/genksyms
-- 
2.34.1

